courses/machine_learning/deepdive/05_review/5_train.ipynb

```diff
@@ -19,19 +19,20 @@
   },
```

Comment:
<p>Should pep8 this block to be </p><p> </p><p>%%writefile babyweight/trainer/task.py</p><p>import argparse</p><p>import json</p><p>import os</p><p> </p><p>from . import model</p><p> </p><p>import tensorflow as tf</p><p> </p><p>if __name__ == "__main__":</p><p>&nbsp;&nbsp;parser = argparse.ArgumentParser()</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--bucket",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="GCS path to data. We assume that data is in \</p><p>&nbsp;&nbsp;&nbsp;&nbsp;gs://BUCKET/babyweight/preproc/",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;required=True</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--output_dir",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="GCS location to write checkpoints and export models",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;required=True</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--batch_size",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="Number of examples to compute gradient over.",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;type=int,</p><p>&nbsp;&nbsp;&nbsp;&nbsp;default=512</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--job-dir",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="this model ignores this field, but it is required by gcloud",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;default="junk"</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--nnsize",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="Hidden layer sizes to use for DNN feature columns -- provide \</p><p>&nbsp;&nbsp;&nbsp;&nbsp;space-separated layers",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;nargs="+",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;type=int,</p><p>&nbsp;&nbsp;&nbsp;&nbsp;default=[128, 32, 4]</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--nembeds",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="Embedding size of a cross of n key real-valued parameters",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;type=int,</p><p>&nbsp;&nbsp;&nbsp;&nbsp;default=3</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--train_examples",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="Number of examples (in thousands) to run the training job over. \</p><p>&nbsp;&nbsp;&nbsp;&nbsp;If this is more than actual # of examples available, it cycles through\</p><p>&nbsp;&nbsp;&nbsp;&nbsp;them. So specifying 1000 here when you have only 100k examples makes \</p><p>&nbsp;&nbsp;&nbsp;&nbsp;this 10 epochs.",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;type=int,</p><p>&nbsp;&nbsp;&nbsp;&nbsp;default=5000</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--pattern",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="Specify a pattern that has to be in input files. For example \</p><p>&nbsp;&nbsp;&nbsp;&nbsp;00001-of will process only one shard",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;default="of"</p><p>&nbsp;&nbsp;)</p><p>&nbsp;&nbsp;parser.add_argument(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;"--eval_steps",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;help="Positive number of steps for which to evaluate model. Default to \</p><p>&nbsp;&nbsp;&nbsp;&nbsp;None, which means to evaluate until input_fn raises an end-of-input \</p><p>&nbsp;&nbsp;&nbsp;&nbsp;exception",</p><p>&nbsp;&nbsp;&nbsp;&nbsp;type=int,</p><p>&nbsp;&nbsp;&nbsp;&nbsp;default=None</p><p>&nbsp;&nbsp;)</p><p> </p><p>&nbsp;&nbsp;# Parse arguments</p><p>&nbsp;&nbsp;args = parser.parse_args()</p><p>&nbsp;&nbsp;arguments = args.__dict__</p><p> </p><p>&nbsp;&nbsp;# Pop unnecessary args needed for gcloud</p><p>&nbsp;&nbsp;arguments.pop("job-dir", None)</p><p> </p><p>&nbsp;&nbsp;# Assign the arguments to the model variables</p><p>&nbsp;&nbsp;output_dir = arguments.pop("output_dir")</p><p>&nbsp;&nbsp;model.BUCKET = arguments.pop("bucket")</p><p>&nbsp;&nbsp;model.BATCH_SIZE = arguments.pop("batch_size")</p><p>&nbsp;&nbsp;model.TRAIN_STEPS = (arguments.pop("train_examples")*1000)/model.BATCH_SIZE</p><p>&nbsp;&nbsp;model.EVAL_STEPS = arguments.pop("eval_steps")</p><p>&nbsp;&nbsp;print ("Will train for {} steps using batch_size={}".format(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;model.TRAIN_STEPS, model.BATCH_SIZE))</p><p>&nbsp;&nbsp;model.PATTERN = arguments.pop("pattern")</p><p>&nbsp;&nbsp;model.NEMBEDS = arguments.pop("nembeds")</p><p>&nbsp;&nbsp;model.NNSIZE = arguments.pop("nnsize")</p><p>&nbsp;&nbsp;print ("Will use DNN size of {}".format(model.NNSIZE))</p><p> </p><p>&nbsp;&nbsp;# Append trial_id to path if we are doing hptuning</p><p>&nbsp;&nbsp;# This code can be removed if you are not using hyperparameter tuning</p><p>&nbsp;&nbsp;output_dir = os.path.join(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;output_dir,</p><p>&nbsp;&nbsp;&nbsp;&nbsp;json.loads(</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;os.environ.get("TF_CONFIG", "{}")</p><p>&nbsp;&nbsp;&nbsp;&nbsp;).get("task", {}).get("trial", "")</p><p>&nbsp;&nbsp;)</p><p> </p><p>&nbsp;&nbsp;# Run the training job</p><p>&nbsp;&nbsp;model.train_and_evaluate(output_dir)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/730/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>Done! (Both for solution and lab)</p>